src/xml_db/parse/mod.rs

```diff
@@ -511,8 +512,16 @@ impl FromXml for CustomData {
             match event {
                 SimpleXmlEvent::Start(name, _) => match &name[..] {
                     "Item" => {
-                        let item = CustomDataItem::from_xml(iterator, inner_cipher)?;
-                        out.items.push(item);
+                        let item = CustomDataItemFromXml::from_xml(iterator, inner_cipher)?;
```

Comment:
@varjolintu if we preserved the `key` field in the `CustomDataItem`, we could simply add a function to `CustomData` that would take care of adding the item to the HashMap. Something as simple as `CustomDataItem::add_item` would enable us to keep the parsing and dumping code unchanged. To only problem that I see is that we might end up with inconsistencies between the key of the hashmap and the key of the `CustomDataItem`, but we could make the `items` field private and provide a few public functions to read and update the items. What do you think?

Reply:
This was actually the whole reason of the PR. If a `HashMap` is not used we must add our own implementation of `add_item`, `remove_item`, `contains` etc. instead of just using what `HashMap` already provides. Sure, we could add some custom implementation under `CustomData`. After this change the usability is closer to how KeePassXC does things. There `CustomDataItem` only contains the value and last modification time.

Even if the database format has a separate variables for key and value, it doesn't mean it should be used directly that way. But sure, it could be a compromise that we keep the `key` in the struct also. It just doesn't have any real meaning, and that means we must add our own `add_item()` because users might forgot to set it.

Reply:
In short, we have a few options:
- Keep the database format intact and all clients using the library must make their own implementations how to handle CustomData.
- We make `items` private and implement our own functions for handling the data.
- Use `HashMap` as the public `items` which allows using it directly, or adding more implementation around it.
- Use `items` as a private `HashMap`, but provide a public function that returns it.

Reply:
I think you're right. The approach you selected is probably the easiest to work with. That way we get both easy access using a key and we can ensure consistency between the key and the custom item.